### ISSUE TITLE
Can't stop/start a WebAppContext with websocket

### ISSUE THREAD

**Issue by janbartel** [eclipse/jetty.project#1216]
Modifying the OneWebAppWithJsp embedded example to include a thread that repetitively stops and starts the webapp context fails:

```
javax.servlet.ServletException: org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter is defined twice for the same context attribute key 'org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter'.  Make sure you have different init-param 'contextAttributeKey' values set
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.setToAttribute(WebSocketUpgradeFilter.java:380)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:354)
	at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:139)
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:725)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:348)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1486)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1443)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:788)
	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:261)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:541)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.embedded.OneWebAppWithJsp$1.run(OneWebAppWithJsp.java:129)

```

Stopping and restarting the same webapp instance should work.

The code I inserted was just before the server.join():

```java       
        Thread t = new Thread() {

            /** 
             * @see java.lang.Thread#run()
             */
            @Override
            public void run()
            {
                while (true)
                {
                    try
                    {
                        Thread.currentThread().sleep(2000);

                        webapp.stop();

                        Thread.currentThread().sleep(2000);

                        webapp.start();

                    }
                    catch (Exception e)
                    {
                        e.printStackTrace();
                    }
                }
            }

        };
        t.setDaemon(true);
        t.start();
```

**Comment by joakime** [comment]
This would be a duplicate / regression of #1114

**Comment by joakime** [comment]
Invalid testcase.

Each time the loop occurs, the filter.destroy() and filter.init() are called multiple times, 1 more for each iteration through the loop.

I added logging to the WSUF.destroy() and WSUF.init() methods to see what's going on, and I see ...

```
2017-01-04 10:35:33.572:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:11.226:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:11.235:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:13.266:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:13.266:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:16.166:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:16.166:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:16.174:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
2017-01-04 10:36:18.198:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:18.198:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:18.199:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:21.086:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:21.086:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:21.086:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:21.095:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
2017-01-04 10:36:23.117:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:23.118:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:23.118:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:23.118:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:45.026:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:45.026:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:45.026:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:45.026:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:45.034:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
2017-01-04 10:36:47.069:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:47.069:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:47.070:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:47.070:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:47.070:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:49.454:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:49.454:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:49.454:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:49.454:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:49.454:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
2017-01-04 10:36:49.470:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .init()
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	|	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.init(WebSocketUpgradeFilter.java:362)
2017-01-04 10:36:51.493:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:51.493:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:51.493:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:51.493:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:51.493:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
2017-01-04 10:36:51.493:INFO:oejws.WebSocketUpgradeFilter:Thread-25: .destroy()
```

**Comment by joakime** [comment]
@janbartel looks like the WebSocketSCI was the ultimate culprit here.

Fixed in commit 0fc8972
